apps/api/src/hyp3_api/validation.py

```diff
@@ -70,31 +70,48 @@ def check_granules_exist(granules, granule_metadata):
         raise GranuleValidationError(f'Some requested scenes could not be found: {", ".join(not_found_granules)}')
 
 
-def check_dem_coverage(granule_metadata):
+def check_dem_coverage(_, granule_metadata):
     bad_granules = [g['name'] for g in granule_metadata if not has_sufficient_coverage(g['polygon'])]
     if bad_granules:
         raise GranuleValidationError(f'Some requested scenes do not have DEM coverage: {", ".join(bad_granules)}')
 
 
-def check_same_burst_ids(granule_metadata):
-    ref_burst_id, sec_burst_id = [granule['name'].split('_')[1] for granule in granule_metadata]
-    if ref_burst_id != sec_burst_id:
+def check_same_burst_ids(job, _):
+    refs = job['job_parameters']['reference']
+    secs = job['job_parameters']['secondary']
+    ref_ids = [ref.split('_')[1] for ref in refs]
+    sec_ids = [sec.split('_')[1] for sec in secs]
+    if len(ref_ids) != len(sec_ids):
+        error_message = (
+            f'Number of reference and secondary scenes must match, got: '
+            f'{len(ref_ids)} references and {len(sec_ids)} secondaries'
+        )
+        raise GranuleValidationError(error_message)
+    if ref_ids != sec_ids:
+        bad_id_index = [ref_ids.index(ref) for (ref, sec) in zip(ref_ids, sec_ids) if ref != sec][0]
+        raise GranuleValidationError(
+            f'Burst IDs do not match for {refs[bad_id_index]} and {secs[bad_id_index]}.'
+        )
+    if len(set(ref_ids)) != len(ref_ids):
+        duplicate_pair_id = [ref_id for ref_id in ref_ids if ref_ids.count(ref_id) > 1][0]
```

Comment:
Note that this is an O(n^2) operation as written. Even if we allowed 100 granule pairs, that would only be 10,000 iterations, so not a real performance concern, but it seems like bad style. I don't see an immediately obvious best way to re-write it, though. Maybe something with [`collections.Counter`](https://docs.python.org/3/library/collections.html#collections.Counter). I'll think more about it.

Reply:
To clarify, it's O(n^2) because it calls `ref_ids.count(ref_id)` (which is a looping operation) for every element in the list. So it's effectively a nested loop.

Reply:
I think I could do a filter() and then use next() so that it stops at the first duplicate pair. It would still be O(n^2) worst case, but probably much better on average. 

Like
```python
filt = lambda ref_id: ref_ids.count(ref_id) > 1
first_dup = next(filter(filt, ref_ids))
```

Although, this might actually be slower because of the function calls. Fun to think about anyway.

Reply:
Meh, I think it's readable enough as-is.

Reply:
Ope, sounds good.